### DISCUSSION TITLE
How to Set Up For Display?

### DISCUSSION THREAD

**Discussion by jpunmd** [syssi/esphome-jk-bms#306] [Q&A]
sensor:
  - platform: jk_bms_ble
    jk_bms_ble_id: bms0
    state_of_charge:
      name: "${name} state of charge"
      id: batterysoc

spi:
  clk_pin: GPIO18
  mosi_pin: GPIO23

font:
  - file: "fonts/Roboto_Medium.ttf"
    id: robotomed
    size: 20

display:
  - platform: ssd1306_spi
    model: "SH1106 128x64"
    cs_pin: GPIO5
    dc_pin: GPIO17
    reset_pin: GPIO16
    lambda: |-
      it.printf(0, 0, id(robotomed), "Battery: %.1F", id(XXXXXXXXXX).state);

What do I fill near the XXXXXXX? I've tried batterysoc, bms0, I've tried appending .cstr().
I've been trying different things for hours and at least 4 different error codes. Can someone smarter than me help me out?

**Comment by syssi** [comment]
Did you find a solution or do you need some support?

**Comment by jpunmd** [comment]
I found the solution. It was the silliest thing, %.0F worked but %.0f did
not despite it being what the documentation on esphome said.

**Reply by syssi** [reply]
That's strange because `%.0f` should work. See https://github.com/syssi/esphome-jbd-bms/discussions/19 f.e.
